storm-multilang/dotnet/src/main/dotnet/Dotnet.Storm.Adapter/Program.cs

```diff
@@ -0,0 +1,113 @@
+            LogLevel level = LogLevel.INFO;
+
+            parser.WithParsed(options =>
+            {
+                className = options.Class;
+                assemblyName = options.Assembly;
+                arguments = options.Arguments;
+
+                // by default TryParse will return TRACE level in case of error
+                Enum.TryParse(options.LogLevel.ToUpper(), out level);
+
+                // if user didn't set TRACE level but we TryParse returned TRACE = parsing exception
+                if (!options.LogLevel.ToLower().Equals("trace") && level == LogLevel.TRACE)
+                {
+                    // reset default log level to INFO
+                    level = LogLevel.INFO;
+                }
+            });
+
+            // Configure logging
+            var repository = LogManager.GetRepository(Assembly.GetEntryAssembly());
+
+            if (File.Exists("log4net.config"))
+            {
+                XmlConfigurator.Configure(repository, new FileInfo("log4net.config"));
+            }
+            else
+            {
+                XmlConfigurator.Configure(repository);
+            }
+
+            // setting up log level
+            Level newLevel = StormAppender.GetLogLevel(level);
+
+            // we want to disable storm logger since storm is not accessible yet
+            StormAppender.Disable(repository, newLevel);
+            Logger.Debug("Storm logger disabled.");
+
+            Logger.Debug($"Current working directory: {Environment.CurrentDirectory}.");
+
+            // we don't want any console logger to be enabled since it is bracking storm multilang protocol
+            // it will not prevent the case other appenders write standard output
+            IAppender[] appenders = repository.GetAppenders();
+            foreach (IAppender appender in appenders)
+            {
+                if (appender.GetType().IsAssignableFrom(typeof(ConsoleAppender)))
+                {
+                    Logger root = ((Hierarchy)repository).Root;
+                    IAppenderAttachable attachable = root as IAppenderAttachable;
+                    attachable.RemoveAppender(appender);
+                    Logger.Debug($"Removing console logger {appender.Name}.");
+                }
+            }
+
+            // Instantiate component
+            Type type = null;
+
+            // className is required option so we don't need to check it for NULL
+            if(!string.IsNullOrEmpty(assemblyName))
+            {
+                Logger.Debug($"Loading assembly: {assemblyName}.");
+                AssemblyName name = new AssemblyName(assemblyName);
+                string path = Path.Combine(Environment.CurrentDirectory, name.Name + ".dll");
+
+                Assembly assembly = Assembly.Load(File.ReadAllBytes(path));
+                type = assembly.GetType(className, true);
+            }
+
+            Component component = (Component)Activator.CreateInstance(type);
```

Comment:
```
try
{
    Component component = (Component)Activator.CreateInstance(type);
}
catch(e)
{
    throw new Exception($"unable to create an instance of type {type} in assembly {assemblyName}", e);
}
```

Reply:
Same thing as in previous comment.

Reply:
I really don't see any reason to hide original exceptions. Activator.CreateInstance will throw an exception in case of error. I don't realize any action the adapter should execute in response to such errors. So I just let a user to see the exception in Storm log.

Reply:
Ok, no problem if we go with the original exception, as long as the error message is comprehensible. This was a suggestion in case it was not - 

Note that the original exception is not hidden, it is passed as a parameter of the new Exception:
` new Exception("message", internalException)`
- it will be linked to the exception you create as an internal one -

Reply:
By default .NET expose only top level message, so if you pass original exception as parameter when you will no see the original message. That's how I understand it.

Reply:
in this case it will be set as InnerException - see https://docs.microsoft.com/en-us/dotnet/api/system.exception.-ctor?view=netframework-4.7.1


```
Exception(String, Exception) | Initializes a new instance of the Exception class with a specified error message and a reference to the inner exception that is the cause of this exception.
-- | --
```